Realm/Realm/Extensions/CollectionExtensions.cs

```diff
@@ -161,6 +161,44 @@ public static IQueryable<T> AsRealmQueryable<T>(this IList<T> list)
             throw new ArgumentException($"{nameof(list)} must be managed by Realm.", nameof(list));
         }
 
+        /// <summary>
+        /// Converts a Realm-backed <see cref="ISet{T}"/> to a Realm-backed <see cref="IQueryable{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the objects contained in the set.</typeparam>
+        /// <param name="set">The set of objects as obtained from a to-many relationship property.</param>
+        /// <returns>A queryable collection that represents the objects contained in the set.</returns>
+        /// <remarks>
+        /// This method works differently from <see cref="Queryable.AsQueryable"/> in that it actually creates
+        /// an underlying Realm query to represent the set. This means that all LINQ methods will be executed
+        /// by the database and also that you can subscribe for notifications even after applying LINQ filters
+        /// or ordering.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// var dogs = owner.Dogs;
+        /// var query = dogs.AsRealmQueryable()
+        ///                 .Where(d => d.Age > 3)
+        ///                 .OrderBy(d =>x d.Name);
+        ///
+        /// var token = query.SubscribeForNotifications((sender, changes, error) =>
+        /// {
+        ///     // You'll be notified only when dogs older than 3 have been added/removed/updated
+        ///     // and the sender collection will be ordered by Name
+        /// });
+        /// </code>
+        /// </example>
+        /// <exception cref="ArgumentException">Thrown if the list is not managed by Realm.</exception>
+        public static IQueryable<T> AsRealmQueryable<T>(this ISet<T> set)
+            where T : RealmObjectBase
+        {
+            if (set is RealmSet<T> realmSet)
+            {
+                return realmSet.ToResults();
+            }
+
+            throw new ArgumentException($"{nameof(set)} must be managed by Realm.", nameof(set));
```

Comment:
You can use `Argument.EnsureType<>` here as well. Feel free to drive-by the IList method.

Reply:
I've changed this with 2003daf and also updated every other function in that class to be consistent.